buildkite/data_source_signed_pipeline_steps.go

```diff
@@ -0,0 +1,244 @@
+					"`jwks_key_id`",
+				),
+				Optional: true,
+			},
+			"jwks": schema.StringAttribute{
+				Description: "The JSON Web Key Set (JWKS) to use for signing. If the `jwks_key_id` is not specified, and the set contains exactly one key, that key will be used.",
+				MarkdownDescription: heredoc.Docf(
+					`
+						The JSON Web Key Set (JWKS) to use for signing.
+						If %s is not specified, and the set contains exactly one key, that key will
+						be used.
+
+						~> **Security Notice** The secret key in this attribute will be stored
+						*unencrypted* in your Terraform state file. This attribute is designed for
+						users that have systems to to securely manage their state files. If you wish
+						to avoid this, use the %s attribute instead.
+
+						See [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517) for more
+						information about the JWKS format.
+					`,
+					"`jwks_key_id`",
+					"`jwks_file`",
+				),
+				Optional:  true,
+				Sensitive: true,
+			},
+			"jwks_key_id": schema.StringAttribute{
+				Description: "The ID of the key in the JSON Web Key Set (JWKS) to use for signing. If this is not specified, and the JWKS contains exactly one key, that key will be used.",
+				MarkdownDescription: heredoc.Doc(
+					`
+						The ID of the key in the JSON Web Key Set (JWKS) to use for signing.
+						If this is not specified, and the key set contains exactly one key, that key
+						will be used.
+
+						See [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517) for more
+						information.
+					`,
+				),
+				Optional: true,
+			},
+			"steps": schema.StringAttribute{
+				Description: "The signed steps in YAML format.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func (s *signedPipelineStepsDataSource) Read(
+	ctx context.Context,
+	req datasource.ReadRequest,
+	resp *datasource.ReadResponse,
+) {
+	data := &signedPipelineStepsDataSource{}
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	p, err := pipeline.Parse(strings.NewReader(data.UnsignedSteps.ValueString()))
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to parse pipeline steps", err.Error())
+		return
+	}
+
+	jwksIsNull := data.JWKS.IsNull()
+	jwksFileIsNull := data.JWKSFile.IsNull()
+
+	var key jwk.Key
+	switch {
+	case jwksIsNull && jwksFileIsNull:
+		resp.Diagnostics.AddError("Exactly one of `jwks` and `jwks_file` needs to be set", "Neither were set.")
+		return
+
+	case !jwksIsNull && !jwksFileIsNull:
+		resp.Diagnostics.AddError("Exactly one of `jwks` and `jwks_file` needs to be set", "Both were set.")
+		return
+
+	case jwksIsNull && !jwksFileIsNull:
+		jwksFile := data.JWKSFile.ValueString()
+		jwksKeyID := data.JWKSKeyID.ValueString()
+
+		key, err = jwkutil.LoadKey(jwksFile, jwksKeyID)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Unable to load JWKS from file",
+				fmt.Sprintf("file: %q, key_id: %q, error: %s", jwksFile, jwksKeyID, err),
+			)
+			return
+		}
```

Comment:
I think all this should be done in validation https://developer.hashicorp.com/terraform/plugin/framework/data-sources/validate-configuration in the schema spec

Then the logic here can just try one then the other

Reply:
Sweet, validation framework is exactly what I needed. I'll give it a shot and holler if I get stuck.